sonar-db/src/main/java/org/sonar/db/component/SnapshotDao.java

```diff
@@ -56,14 +54,8 @@ public SnapshotDto selectOrFailById(DbSession session, long id) {
     return value;
   }
 
-  public List<SnapshotDto> selectByIds(final DbSession dbSession, List<Long> snapshotIds) {
-    return executeLargeInputs(snapshotIds, new Function<List<Long>, List<SnapshotDto>>() {
-      @Nonnull
-      @Override
-      public List<SnapshotDto> apply(@Nonnull List<Long> input) {
-        return mapper(dbSession).selectByIds(input);
-      }
-    });
+  public List<SnapshotDto> selectByIds(DbSession dbSession, List<Long> snapshotIds) {
+    return executeLargeInputs(snapshotIds, (input) -> mapper(dbSession).selectByIds(input));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1612)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove the parentheses around the "input" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1611)